src/base/datetime/calendar/header/dropdown/month/index.ts

```diff
@@ -0,0 +1,229 @@
+import { BaseDateTimeListbox, Item } from "../../../../listbox";
+import { en } from "../../../../resource/locale";
+import { getToggleIconSvgTemplate, getLocale } from "../../ultils";
+
+export class BaseDateTimeMonthDropdown extends KucBase {
+  @property({ type: String }) language = "en";
+  @property({ type: Number }) month = 1;
+
+  @state()
+  private _menuVisible = false;
+  private _locale = en;
+  private _monthLabel = "";
+  private _GUID = generateGUID();
+  private _menuItems: Item[] | undefined;
+  private _maxHeight = 1000;
+
+  @query(".kuc-base-datetime-month-dropdown__toggle")
+  private _toggleEl!: HTMLButtonElement;
+
+  @query(".kuc-base-datetime-month-dropdown__listbox")
+  private _menuEl!: BaseDateTimeListbox;
+
+  update(changedProperties: PropertyValues) {
+    if (changedProperties.has("language")) {
+      this._locale = getLocale(this.language);
+      this._menuItems = this._getMenuItems();
+    }
+    if (changedProperties.has("month")) {
+      this._monthLabel = this._getMonthLabel();
+    }
+    super.update(changedProperties);
+  }
+
+  render() {
+    return html`
+      ${this._getStyleTagTemplate()}
+      <button
+        class="kuc-base-datetime-month-dropdown__toggle"
+        aria-haspopup="true"
+        aria-labelledby="${this._GUID}-label ${this._GUID}-toggle"
+        @mouseup="${this._handleMouseUpDropdownToggle}"
+        @mousedown="${this._handleMouseDownDropdownToggle}"
+        @click="${this._handleClickDropdownMonthToggle}"
+        @blur="${this._handleBlurDropdownMonthToggle}"
+        @keydown="${this._handleKeyDownMonthToggle}"
+      >
+        <span class="kuc-base-datetime-month-dropdown__toggle__label"
+          >${this._monthLabel}</span
+        >
+        <span class="kuc-base-datetime-month-dropdown__toggle__icon"
+          >${getToggleIconSvgTemplate()}
+        </span>
+      </button>
+      <kuc-base-datetime-listbox
+        .items="${this._menuItems || []}"
+        .value="${this.month.toString()}"
+        .maxHeight="${this._maxHeight}"
+        class="kuc-base-datetime-month-dropdown__listbox"
+        @kuc:calendar-listbox-click="${this._handleChangeMenu}"
+        aria-hidden="${!this._menuVisible}"
+        ?hidden="${!this._menuVisible}"
+      >
+      </kuc-base-datetime-listbox>
+    `;
+  }
+
+  private _getStyleTagTemplate() {
+    return html`
+      <style>
+        .kuc-base-datetime-month-dropdown__toggle {
+          position: relative;
+          box-sizing: border-box;
+          height: 32px;
+          padding: 0 14px 0 8px;
+          line-height: 30px;
+          overflow: hidden;
+          background-color: white;
+          border: 1px solid transparent;
+          cursor: pointer;
+        }
+        .kuc-base-datetime-month-dropdown__toggle__icon {
+          flex: none;
+          width: 38px;
+          height: 38px;
+        }
+      </style>
+    `;
+  }
+
+  private _handleClickDropdownMonthToggle(event: MouseEvent) {
```

Comment:
Delete the unnecessary MouseEvent.
```suggestion
  private _handleClickDropdownMonthToggle() {
```